app/form_persisters/form_persister.rb

```diff
@@ -25,37 +20,63 @@ def save_all(forms:)
     end
   end
 
-  private
+  def instance(form)
+    Instance.new(form: form, form_persister: self)
+  end
 
-    def before_save(form:)
-      create_files(form: form)
+  class Instance
+    attr_reader :form, :form_persister
+    delegate :adapter, :storage_adapter, to: :form_persister
+    delegate :persister, :query_service, to: :adapter
```

Comment:
Excellent perspective. Is `Instance` something that should be a private_constant?